module/polls/helper.js

```diff
@@ -0,0 +1,631 @@
+                pollDocument.result = result;
+                
+                let createPollResult = await database.models.polls.create(pollDocument)
+
+                let link = await gen.utils.md5Hash(userId + "###" + createPollResult._id);
+
+                await database.models.polls.updateOne
+                (
+                    { _id : createPollResult._id },
+                    {
+                        $set : { link : link}
+                    }
+                )
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_CREATED,
+                    data: {
+                        link : appsPortalBaseUrl + appName + messageConstants.common.TAKE_POLL + link
+                    }
+                });
+                
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+    static list(userId= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if(userId == ""){
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                let pollsList = await this.pollDocuments
+                (
+                    {
+                        createdBy: userId,
+                        isDeleted: false
+                    },
+                    [
+                        "name"
+                    ],
+                    {createdAt: 1}
+                )
+
+                if (!pollsList.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: pollsList
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+     * Delete an poll.
+     * @method
+     * @name delete
+     * @param {String} pollId - pollId 
+     * @returns {String} - message.
+     */
+
+    static delete(pollId= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if(pollId == ""){
+                    throw new Error(messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK);
+                }
+
+                await database.models.polls.updateOne
+                (
+                    { _id: pollId },
```

Comment:
@deepakr123 - Check only the creator of the poll can mark his pole as deleted or inactive etc.

Reply:
resolved